x-pack/plugins/apm/server/lib/helpers/setup_request.test.ts

```diff
@@ -117,102 +125,96 @@ describe('setupRequest', () => {
     const { mockContext, mockRequest } = getMockRequest();
     const { internalClient } = await setupRequest(mockContext, mockRequest);
     await internalClient.search({
-      index: 'apm-*',
+      index: ['apm-*'],
```

Comment:
I don't think the internalClient should be able to query any index but instead scoped to a specific (system) index.

Reply:
Are we using this anywhere yet or is that WIP?

Reply:
I think we agreed to for now only create an APM-specific client, and keep the internal client as-is. We're using the internalClient for `.apm-custom-link` and `.apm-agent-configuration` IIRC.

Reply:
We can do that - I just think it would be a little cleaner to separate it, and safer in the sense that the consumer cannot mistakenly read from the wrong index.